AU: Autenticación del usuario

fragonlug opened the pull request:
Se han creado los modelos correspondientes para el usuario de la aplicación y se ha implementado el login y logout de la aplicación.

alfgonfer committed:
Realizado login y logout de usuario-#6

Se han realizado el login y logout de usuario junto a unas pruebas.

fragonlug committed:
Creado los modelos de UsuarioPerfil, ContadorVida y Premiun-#6

Se han creado las clases correspondientes para el usuario de la aplicación.

alfgonfer committed:
Añadido migrate al travis.yml-#6

Tras los últimos cambios realizados sobre el modelo, falla la build de travis y procedemos a arreglarlo.

alfgonfer committed:
Cambiado nombre de database-#6

Se ha cambiado el nombre de la base de datos que se crea para pasar las pruebas de travis.

fragonlug committed:
Añadido el archivo generado tras realizar el migrate

Se ha añadio el archivo generado por el migrate, ajustado el .travis.yml y eliminado la excepción del gitignore.

ferruirob commented:
La rama tiene errores detectados por codacy, hasta que no pase los pase no puedo aceptar el merge.  Arregla los errores por favor. Gracias [https://app.codacy.com/gh/youRoom-ISPP/youRoom/pullRequest?prid=7131144#newIssuesView]

fragonlug committed:
Realizado los ajustes en relación a los cometarios recibidos por Codacy

Los revisores de la pull request han identificado issues ofrecidas por Codacy y han recomendado su correción.

fragonlug commented:
> La rama tiene errores detectados por codacy, hasta que no pase los pase no puedo aceptar el merge. Arregla los errores por favor. Gracias [https://app.codacy.com/gh/youRoom-ISPP/youRoom/pullRequest?prid=7131144#newIssuesView]

Se han realizado los cambios pertinentes en respuesta a estas incidencias.

fragonlug committed:
Realizado ajustes en los endpoints de las apps-#6

Se han realizado los ajustes pertinentes para corresponder al flujo ideado para la página.

fragonlug committed:
Añadido el html del login

Se ha añadido el html del login para el correcto funcionamiento de las pruebas.

fragonlug committed:
Merge branch 'develop' into AU-#6

mrf1989 commented on youroom/usuario/urls.py:
```diff
@@ -1,6 +1,11 @@
-]
\ No newline at end of file
+    path('', LoginView.as_view(), name='login'),
+    path('', include(timeline_urls), name='home'),
```
@fragonlug  Es importante que el name no tenga una referencia errónea o que no corresponde con los términos utilizados en el proyecto. En este caso, name tendría que tener el valor "timeline".

mrf1989 commented on youroom/youroom/settings.py:
```diff
@@ -132,3 +132,6 @@
 STATICFILES_DIRS = [
     os.path.join(BASE_DIR, 'static'),
 ]
+LOGIN_REDIRECT_URL = 'home'
```
@fragonlug Lo mismo digo, la redirección del login no se hace a la "home", que no tenemos como tal, sino al "timeline".

mrf1989 commented on youroom/usuario/tests.py:
```diff
@@ -1,3 +1,31 @@
+        response = self.client.post('', answers)
+        self.assertEqual(response.status_code, 200)
+
+    
```
@fragonlug Estaría bien contar con un test de caso negativo.

mrf1989 reviewed (changes requested):
@fragonlug Te dejo algunos comentarios en sus correspondientes líneas que tendrían que resolverse para no arrastrar cosas a la línea de desarrollo.

Las hemos hablado antes.

fragonlug committed:
Se han realizado los cambios en respuesta a la review recibida-#6

Se han modificado los endpoint y se ha añadio un test negativo de login.

fragonlug committed:
Cambios aplicados para resolver los conflictos-#6

Este es el resultado de la resolución de conflictos entre la rama AU-#6 y develop.

fragonlug commented:
> @fragonlug Te dejo algunos comentarios en sus correspondientes líneas que tendrían que resolverse para no arrastrar cosas a la línea de desarrollo.
> 
> Las hemos hablado antes.

Se han tomado en cuenta las anotaciones y se han realizado los cambios pertinentes.

mrf1989 reviewed (approved):
Perfecto! Por mi parte, nada más que añadir.

ferruirob commented:
Revisado y todo correcto. Apruebo el merge request por mi parte también.
